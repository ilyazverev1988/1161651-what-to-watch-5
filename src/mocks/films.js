export default [
  {
    preview: `img/aviator.jpg`,
    nameFilm: `Aviator`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Thrillers`,
    releaseYear: 2014,
    descriptionFilm: `Friendship, metamorphosis, and adventure.`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Everything we do is connected with zen: sex, bliss, moonlight, sorrow.`,
    producer: `Heu, orexis!`,
    duration: `1h 39m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/dardjeeling-limited.jpg`,
    nameFilm: `Dardjeeling limited`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2016,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/bohemian-rhapsody.jpg`,
    nameFilm: `Bohemian rhapsody`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    nameFilm: `Fantastic beasts the crimes of grindelwald`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2018,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 50m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/johnny-english.jpg`,
    nameFilm: `Johnny english`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2019,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 9.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 30m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/macbeth.jpg`,
    nameFilm: `Macbeth`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2020,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 5.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 33m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/no-country-for-old-men.jpg`,
    nameFilm: `No country for old men`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2017,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 44m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/war-of-the-worlds.jpg`,
    nameFilm: `War of the worlds`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 54m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
];
