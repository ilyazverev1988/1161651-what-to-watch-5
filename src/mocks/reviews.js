export default [
  {
    film: `Aviator`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere1.`,
        filmScore: `1`,
        nameUser: `Ubi est albus1?`,
        dateReview: `December 24, 2018`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere2.`,
        filmScore: `2`,
        nameUser: `Ubi est albus2?`,
        dateReview: `December 20, 2018`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere3.`,
        filmScore: `3`,
        nameUser: `Ubi est albus3?`,
        dateReview: `December 21, 2018`
      }
    ]
  },
  {
    film: `Dardjeeling limited`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `4`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2018`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 20, 2019`
      },
    ]
  },
  {
    film: `Bohemian rhapsody`,
    review: [
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 13, 2018`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `1`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 21, 2011`
      }
    ]
  },
  {
    film: `Fantastic beasts the crimes of grindelwald`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2010`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 10, 2010`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `4`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 14, 2015`
      }
    ]
  },
  {
    film: `Johnny english`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `2`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 12, 2010`
      },
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `3`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 3, 2010`
      }
    ]
  },
  {
    film: `Macbeth`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 5, 2010`
      }
    ]
  },
  {
    film: `No country for old men`,
    review: [
      {
        id: 2,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 6, 2010`
      },
      {
        id: 3,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 26, 2010`
      }
    ]
  },
  {
    film: `War of the worlds`,
    review: [
      {
        id: 1,
        textReview: `Courage is not one in shangri-la, the country of result, or earth, but everywhere.`,
        filmScore: `5`,
        nameUser: `Ubi est albus ionicis tormento?`,
        dateReview: `December 13, 2015`
      }
    ]
  },
];
