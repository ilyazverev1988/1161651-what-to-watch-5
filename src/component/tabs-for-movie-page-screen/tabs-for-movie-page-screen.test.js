import React from "react";
import renderer from "react-test-renderer";
import TabsForMoviePageScreen from "./tabs-for-movie-page-screen";
import films from "../../mocks/films";
import reviews from "../../mocks/reviews";

const noop = () => {};

describe(`Should TabsForMoviePageScreen render correctly`, () => {
  it(`With active Details`, () => {
    const tree = renderer
      .create(
          <TabsForMoviePageScreen
            film={films[1]}
            reviews={reviews}
            handleClickOverview={noop}
            handleClickDetails={noop}
            handleClickReviews={noop}
            Details={true}
            Overview={false}
            Reviews={false}
          />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With active Overview`, () => {
    const tree = renderer
      .create(
        <TabsForMoviePageScreen
          film={films[1]}
          reviews={reviews}
          handleClickOverview={noop}
          handleClickDetails={noop}
          handleClickReviews={noop}
          Details={false}
          Overview={true}
          Reviews={false}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With active Reviews`, () => {
    const tree = renderer
      .create(
        <TabsForMoviePageScreen
          film={films[1]}
          reviews={reviews}
          handleClickOverview={noop}
          handleClickDetails={noop}
          handleClickReviews={noop}
          Details={false}
          Overview={false}
          Reviews={true}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
