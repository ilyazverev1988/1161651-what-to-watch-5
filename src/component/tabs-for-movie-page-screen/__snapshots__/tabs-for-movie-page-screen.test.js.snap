// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TabsForMoviePageScreen render correctly With active Details 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Wes Andreson
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Bill Murray
          <br />
          Jude Law
          
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1h 32m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Thrillers
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2014
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Should TabsForMoviePageScreen render correctly With active Overview 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        3
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      Friendship, metamorphosis, and adventure.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Wes Andreson
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Bill Murray, Jude Law and other
      </strong>
    </p>
  </div>
</div>
`;

exports[`Should TabsForMoviePageScreen render correctly With active Reviews 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item  "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Mollie
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              October 27, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3.2
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kent
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              October 27, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          5
        </div>
      </div>
    </div>
  </div>
</div>
`;
